src/actions/LogActions.tsx

```diff
@@ -197,40 +221,54 @@ function prepareLogs(text: string): ThunkAction<Promise<void>> {
         logOutput.loggedInUser.actions.push(actionData)
       }
     }
-    logOutput.data += `App version: ${packageJson.version}
-App build: ${getVersion()}.${getBuildNumber()}
-os: ${Platform.OS} ${Platform.Version}
-device: ${getBrand()} ${getDeviceId()}
-`
+    const appInfo = `
+      App version: ${packageJson.version}
+      App build: ${getVersion()}.${getBuildNumber()}
+      os: ${Platform.OS} ${Platform.Version}
+      device: ${getBrand()} ${getDeviceId()}
+    `
+
+    logOutput.data += appInfo
 
     //
     // Send log output
     //
 
-    await logWithType('activity', 'SENDING ACTIVITY LOGS WITH MESSAGE: ' + text)
-      .then(async () => await readLogs('activity'))
-      .then(async logs => {
-        activityOutput.data += logs || ''
-        return sendLogs(activityOutput)
-      })
-      .catch(e => {
-        throw new Error(`${s.strings.settings_modal_send_logs_failure} activity logs code ${e.message}`)
-      })
-
-    await log('SENDING INFO LOGS WITH MESSAGE: ' + text)
-      .then(async () => await readLogs('info'))
-      .then(async logs => {
-        logOutput.data += logs || ''
-        return sendLogs(logOutput)
-      })
-      .catch(e => {
-        throw new Error(`${s.strings.settings_modal_send_logs_failure} info logs code ${e.message}`)
-      })
+    try {
+      const activityLogs = await readLogs('activity')
+
+      activityOutput.data += activityLogs != null ? activityLogs : ''
+      const activityPayload = JSON.stringify(activityOutput, null, 2)
+      const safeToSend = !SENSITIVE_KEY_REGEX.test(activityPayload)
+      if (send) {
+        if (safeToSend) await sendLogs(activityPayload)
+        else throw new Error('⚠ ' + s.strings.settings_modal_send_unsafe)
+      }
+    } catch (error: any) {
+      throw new Error(`${s.strings.settings_modal_send_logs_failure} (activity logs):\n\n${error.message}`)
+    }
```

Comment:
This try / catch logic would move to the modal, since that's where translated strings belong.

Reply:
Moved there now.